core/state_processor.go

```diff
@@ -74,17 +78,22 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.DB, cfg vm.C
 		}
 		receipts = append(receipts, receipt)
 		if cxReceipt != nil {
-			cxs = append(cxs, cxReceipt)
+			outcxs = append(outcxs, cxReceipt)
 		}
 		allLogs = append(allLogs, receipt.Logs...)
 	}
+
+	for _, cx := range block.IncomingReceipts() {
```

Comment:
Add a note: incomingReceipts should always be processed after transactions (to be consistent with the block proposal)

Reply:
comments added